src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterHelper.cs

```diff
@@ -16,16 +17,29 @@ public partial class BinaryFormatterTests
     {
         private static void CheckForAnyEquals(object obj, object deserializedObj)
         {
-            if (obj != null && deserializedObj != null)
+            try
+            {
+                Assert.True(CheckEquals(obj, deserializedObj));
+            }
+            catch (Exception)
+            {
+                Console.WriteLine("Error during equality check of type " + obj?.GetType()?.FullName);
```

Comment:
@danmosemsft this is wrapping assertion because Assert gives really not readable message